tests/specs/advanced.test.ts

```diff
@@ -168,6 +168,74 @@ describe(`Advanced`, () => {
     });
   });
 
+  describe(`Tokenization`, () => {
+    const cli = Cli.from([
+      class extends Command {
+        static paths = [[`main`]];
+
+        foo = Option.Boolean(`-f,--foo`);
+        bar = Option.Boolean(`-b,--bar`);
+
+        hello = Option.String(`--hello`);
+
+        arg1 = Option.String({required: false});
+        arg2 = Option.Rest();
```

Comment:
Optional positionals and rests can't be declared together as they are both `arity.extra`.

`addPositional` does check whether a `NoLimits` is already declared before adding an optional positional, but `addRest` doesn't check whether a `string[]` extra already exists, so unfortunately there's no error to indicate this.

Reply:
Interesting, that sounds like a bug, I'd expect it to work - I added a skipped test, I may try to look into it at some point.